### PR TITLE
Fix for JsonUtils stringToDate to support Microsoft Dates with TimeZones

### PR DESCRIPTION
**Pull request by cmelchior**
Replaces #1533. This PR fixes the last couple of issues so it is ready to be merged.

@realm/java

### CODE REVIEW

File: realm/src/androidTest/java/io/realm/RealmJsonTest.java [hunk 1]
```diff
@@ -121,6 +124,26 @@ public void testCreateObjectFromJson_dateAsString() throws JSONException {
         assertEquals(new Date(1000), obj.getColumnDate());
     }
 
+    public void testCreateObjectFromJson_dateAsStringTimeZone() throws JSONException {
+        JSONObject json = new JSONObject();
+
+        // Oct 03 2015 14:15
```
Comment (zaki50) [inline realm/src/androidTest/java/io/realm/RealmJsonTest.java]: 14:45?

File: realm/src/main/java/io/realm/internal/android/JsonUtils.java [hunk 2]
```diff
@@ -17,18 +17,21 @@
...
+import java.util.Calendar;
 import java.util.Date;
+import java.util.GregorianCalendar;
 import java.util.regex.Matcher;
 import java.util.regex.Pattern;
 
 public class JsonUtils {
 
-    private static Pattern jsonDate = Pattern.compile("/Date\\((\\d*)\\)/");
+    private static Pattern jsonDate = Pattern.compile("Date\\((\\d*)([+-]\\d*)?\\)");
```
Comment (zaki50) [inline realm/src/main/java/io/realm/internal/android/JsonUtils.java]: `([+-]\\d*)` can be `(?:[+-]\\d*)` since we don't need to capture it.

**Comment by kneth** [comment]
Look good to me.

**Comment by zaki50** [comment]
:+1:

**Pull request merged**